src/Logger.ts

```diff
@@ -0,0 +1,68 @@
+import { createLogger, format, Logger as WinstonLogger, transports } from 'winston'
+const { colorize, combine, printf, json, timestamp } = format
+
+const consoleFormat = printf(({ level, message, timestamp, handlerId }) => {
+  return `${timestamp} ${handlerId ? 'event' : 'runtime'} ${level} ${message}`
+})
+
+const filterLog = (withHandlerLogs: boolean) =>
+  format((info) =>
+    info.handlerId ? (withHandlerLogs ? info : false) : withHandlerLogs ? false : info,
+  )
+
+const loggerOptions = {
+  level: 'info',
+  format: combine(timestamp(), json()),
+  exitOnError: false,
+  transports: [
+    // - Write all logs with level `error` and below to `error.log`
+    new transports.File({ filename: './logs/error.log', level: 'error' }),
+    // - Write all logs with level `info` and below to `combined.log`
+    new transports.File({ filename: './logs/combined.log' }),
+    new transports.File({ filename: './logs/handlers.log', format: filterLog(true)() }),
+    new transports.Console({
+      format: combine(
+        // filterLog(false)(),
+        colorize({ level: true, message: true }),
+        consoleFormat,
+      ),
+    }),
+  ],
+}
+
+export default class Logger {
+  private readonly logger: WinstonLogger
+  private consoleFunctions = {
+    log: console.log,
+    info: console.info,
+    warn: console.warn,
+    error: console.error,
+    debug: console.debug,
+  }
+
+  constructor(options = loggerOptions) {
+    this.logger = createLogger(options)
+  }
+
+  redirectConsoleToLogger(): void {
```

Comment:
I do not want to highjack console in Reshuffle fabric. as a library it is an anti pattern.

Reply:
removed